haystack/document_stores/deepsetcloud.py

```diff
@@ -61,12 +62,32 @@ def __init__(
         self.client = DeepsetCloud.get_index_client(
             api_key=api_key, api_endpoint=api_endpoint, workspace=workspace, index=index
         )
+        # Check if index exists
+        pipeline_client = DeepsetCloud.get_pipeline_client(
+            api_key=api_key, api_endpoint=api_endpoint, workspace=workspace
+        )
 
-        index_info = self.client.info()
-        indexing_info = index_info["indexing"]
-        if indexing_info["pending_file_count"] > 0:
+        deployed_pipelines = set(
+            pipe["name"]
+            for pipe in pipeline_client.list_pipeline_configs(workspace=workspace)
+            if pipe["status"] == "DEPLOYED"
+        )
+        self.index_exists = index in deployed_pipelines
+
+        if self.index_exists:
+            index_info = self.client.info()
+            indexing_info = index_info["indexing"]
+            if indexing_info["pending_file_count"] > 0:
+                logger.warning(
+                    f"{indexing_info['pending_file_count']} files are pending to be indexed. "
+                    f"Indexing status: {indexing_info['status']}"
+                )
+        else:
             logger.warning(
-                f"{indexing_info['pending_file_count']} files are pending to be indexed. Indexing status: {indexing_info['status']}"
+                f"You are using a DeepsetCloudDocumentStore with an index that does not exist on deepset Cloud. "
+                f"This document store will always return empty responses. To add files to your index, first upload "
+                f"files and publish&deploy your pipeline. "
+                f"DC will automatically index your files for you."
```

Comment:
I would convert that into an info as this is a common case. Additionally I'd rather add a hint on how to create the index on deepset cloud: 
```suggestion
                f"This document store will always return empty responses. This is especially useful if you want to create a new pipeline within deepset Cloud.\n"
                f"In order to create a new pipeline on deepset Cloud, take the following steps: \n"
                f"  - create query and index pipelines using this DocumentStore\n"
                f"  - call `Pipeline.save_to_deepset_cloud()` passing the pipelines and a `pipeline_config_name`\n"
                f"  - call `Pipeline.deploy_on_deepset_cloud()` passing the `pipeline_config_name`"
```

Reply:
Done.